File: yurt-system/yurt-system-server/src/main/java/com/github/liulus/yurt/system/repository/RoleRepository.java
package com.github.liulus.yurt.system.repository;

import com.github.liulus.yurt.jdbc.JdbcRepository;
import com.github.liulus.yurt.jdbc.annotation.Select;
import com.github.liulus.yurt.system.model.entity.Role;

import java.util.List;

/**
 * @author liulu
 * @version 1.0
 * created_at 2020/5/26
 */
public interface RoleRepository extends JdbcRepository<Role> {

    @Select(leftJoin = "user_role ur on role.id = ur.role_id", where = "ur.user_id = :param")
    List<Role> selectByUserId(Long userId);
}

File: yurt-system/yurt-system-server/src/main/kotlin/com/github/liulus/yurt/system/repository/RoleRepository.kt
package com.github.liulus.yurt.system.repository

import com.github.liulus.yurt.jdbc.JdbcRepository
import com.github.liulus.yurt.jdbc.annotation.Select
import com.github.liulus.yurt.system.model.entity.Role

/**
 * @author liulu
 * @version 1.0
 * created_at 2020/5/26
 */
interface RoleRepository : JdbcRepository<Role?> {
    @Select(leftJoin = "user_role ur on role.id = ur.role_id", where = ["ur.user_id = :param"])
    fun selectByUserId(userId: Long): List<Role>
}